frontend/src/components/AddClassModal.tsx

```diff
@@ -220,20 +199,49 @@ export class AddClassModal extends React.Component<
         {this.state.queuedCourses.map(
           (course: ScheduleCourse, index: number) => {
             return (
-              <QueuedClass>
-                <p style={{ width: 90 }}>{course.subject + course.classId}</p>
-                <p>{course.name}</p>
-                <XButton
-                  onClick={this.removeQueuedClass.bind(this, index)}
-                ></XButton>
-              </QueuedClass>
+              <NonDraggableClassBlock
+                course={course}
+                onDelete={this.removeQueuedClass.bind(this, index)}
+              />
             );
           }
         )}
       </QueuedClassesWrapper>
     );
   }
 
+  renderSubjectTextField() {
+    return (
+      <TextField
+        id="outlined-basic"
+        label="Subject"
+        variant="outlined"
+        value={this.state.formSubject}
+        onChange={this.handleSubjectChange.bind(this)}
+        placeholder="CS"
+        error={this.state.formSubject.length === 0}
+        style={{ marginTop: 36, minWidth: 326 }}
+        color="secondary"
+      />
+    );
+  }
+
+  renderClassIDTextField() {
+    return (
+      <TextField
+        id="outlined-basic"
+        label="Class ID"
+        variant="outlined"
+        value={this.state.formClassId}
+        onChange={this.handleClassIdChange.bind(this)}
+        placeholder="3500"
+        error={this.state.formClassId.length === 0}
+        style={{ marginTop: 12, marginBottom: 12, minWidth: 326 }}
+        color="secondary"
```

Comment:
I'd say don't use secondary since we don't use that in our color scheme anymore. You can leave it as default or use this color -> #EB5757 (which is the same as the NextButton) (same with the subject text field)